src/components/WorkBuddies/WorkBuddies.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {
+  Fragment,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+  useMemo,
+} from "react";
+import "./work-buddies.css";
+import classNames from "classnames";
+import { Buddy } from "./BuddyInterface";
+import { BuddiesContext } from "../BuddiesContext";
+import MyWorkBuddy from "./MyWorkBuddy";
+
+const WorkBuddies = () => {
+  const [buddies, setBuddies] = useState<Buddy[]>([]);
+  const [myBuddy, setMyBuddy] = useState<Buddy>();
+  const [isBuddyChosen, setIsBuddyChosen] = useState<boolean>(false);
+  //useRef
+  const nameRef = useRef<HTMLInputElement>(null);
+
+  //useContext
+  const { setData } = useContext(BuddiesContext);
+
+  //useCallback
+  const fetchBuddies = useCallback(async () => {
+    try {
+      const response = await fetch(
+        "https://pokeapi.co/api/v2/pokemon?limit=100"
+      );
+      const data = await response.json();
+
+      const randomIndexes: number[] = [];
+
+      while (randomIndexes.length < 9) {
+        const randomNumber = Math.floor(Math.random() * 100);
+        if (!randomIndexes.includes(randomNumber)) {
+          randomIndexes.push(randomNumber);
+        }
+      }
+
+      const todaysBuddies: Buddy[] = randomIndexes.map((index: number) => {
+        let urlSplitted = data.results[index].url.split("/");
+        let newUrl = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${
+          urlSplitted[urlSplitted.length - 2]
+        }.png`;
+
+        return { name: data.results[index].name, url: newUrl };
+      });
+
+      setBuddies(todaysBuddies);
+    } catch (error) {
+      console.log(error);
+    }
+  }, [setBuddies]);
+
+  //useMemo
+  const displayBuddies = useMemo(() => {
+    return buddies.map((buddy: Buddy, index) => {
+      return (
+        <div
+          className={classNames("buddy-small", { selected: myBuddy === buddy })}
+          key={index}
+          onClick={() => {
+            setMyBuddy(buddy);
+          }}
+        >
+          <img src={buddy.url} alt="buddy-image" />
+        </div>
+      );
+    });
+  }, [buddies, myBuddy]);
+
+  const onChooseOne = () => {
+    if (nameRef.current && nameRef.current.value) {
+      setMyBuddy((prev) => ({
+        ...(prev as Buddy),
+        name: nameRef.current!.value,
+        isMine: true,
+      }));
+    }
+
+    if (myBuddy === undefined) {
+      return;
+    }
+
+    const updatedBuddies = buddies;
```

Comment:
This doesn't work the way you are expecting.
JS works with references.
If you want to create a new copy of the array so that you want to mutate it, you should do something like so:
```suggestion
    const updatedBuddies = [...buddies];
```